client/components/editor/teaching-elements/Assessment/ImageMultipleChoice.vue

```diff
@@ -0,0 +1,326 @@
+<template>
+  <div :class="{ disabled }">
+    <h5>Answers</h5>
+    <span @click="addAnswer" class="btn btn-link mdi mdi-plus pull-right"></span>
+    <ul>
+      <li
+        v-for="(answer, index) in answers"
+        :key="getUniqueIndex(index)"
+        style="text-align: start;">
+        <span>
+          <input
+            :checked="correct.includes(index)"
+            :disabled="disabled"
+            @change="toggleAnswer(index)"
+            type="checkbox">
+        </span>
+        <span :class="{ 'has-error': getErrorMessages(index).length > 0 }">
+          <div class="image-container">
+            <img :src="answer || './assets/img/no-image.png'" class="image-content">
+          </div>
+          <input
+            :data-index="index"
+            type="file"
+            :disabled="disabled"
+            @change="updateAnswer"
+            class="form-control image-input"
+            :accept="imageInputType">
+          <span
+            v-for="(message, index) in getErrorMessages(index)"
+            :key="index"
+            class="error-msg">
+            {{ message }}
+          </span>
+        </span>
+        <span @click="removeAnswer(index)" class="mdi mdi-close control"></span>
+      </li>
+    </ul>
+  </div>
+</template>
+
+<script>
+import cloneDeep from 'lodash/cloneDeep';
+import range from 'lodash/range';
+import { blobToDataURL } from 'blob-util';
+
+const maxImageDimension = 256;
+
+const formAlert = (message) => {
+  return {
+    type: 'alert-danger',
+    text: message
+  };
+};
+
+const getImageDimensions = function (imageDataUrl) {
+  return new Promise(resolve => {
+    let i = document.createElement('img');
+    i.onload = () => resolve({ width: i.width, height: i.height });
+    i.src = imageDataUrl;
+  });
+};
+
+class ImageValidationError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = 'ImageValidationError';
+  }
+}
```

Comment:
Please read [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error#Custom_Error_Types) first. `name` prop is pointless because you should use `instanceof` check instead...